ovs-dpdk.md

```diff
@@ -0,0 +1,227 @@
+ 
+The network card must be on the supported-vendor matrix, located here: <http://dpdk.org/doc/nics>
+
+
+## Configure OVS and DPDK on the Host
+
+* Make sure VT-d is enabled in BIOS
+* When adding a new host via oVirt GUI, add the following parameters to Kernel Command line:  “iommu=pt intel_iommu=on default_hugepagesz=1GB hugepagesz=1G hugepages=N”. 
+
+
+### Install dpdk packages from fast data path repo:
+
+```bash
+# yum install dpdk
+
+```
+ 
+### Bind NICS to DPDK:
+
+In order to run a DPDK application, a suitable uio module should be loaded into kernel memory.
+Available modules are: uio_pci_generic, igb_uio and vfio-pci.
+In many cases, the standard uio_pci_generic module included in the Linux kernel can provide the uio capability.
+For some devices which lack support for legacy interrupts, igb_uio should be used, but it’s an out-of-tree kernel module.
+Vfio is preferred to uio drivers in latest DPDK versions in platforms that support VFIO.
+
+In order to support VFIO when running OVS-DPDK as non privileged user, proper permissions should be set up:
+
+``` bash
+# modprobe vfio-pci
+# sudo /usr/bin/chmod a+x /dev/vfio
+# sudo /usr/bin/chmod 0666 /dev/vfio/*
+```
+
+```bash
+# sudo modprobe vfio-pci
+```
+
+All ports that you would like to run DPDK application should be bound to one of the modules before OVS is started. DPDK will ignore all devices which are not bound to one of the drivers above.
+ 
+Show status of all ports on a system:
+
+```bash
+Unbind a port from Linux control:
+
+# dpdk-devbind -s
+ 
+Network devices using DPDK-compatible driver
+============================================
+<none>
+ 
+Network devices using kernel driver
+===================================
+0000:02:00.0 '82576 Gigabit Network Connection' if=enp2s0f0 drv=igb unused= 
+0000:02:00.1 '82576 Gigabit Network Connection' if=enp2s0f1 drv=igb unused= 
+ 
+Other network devices
+=====================
+<none>
+ 
+Crypto devices using DPDK-compatible driver
+===========================================
+<none>
+ 
+Crypto devices using kernel driver
+==================================
+<none>
+ 
+Other crypto devices
+====================
+<none>
+
+```
+
+Unbind a port from Linux control:
+
+```
+# dpdk-devbind -u <pci_addr>
+
+```
+
+Bind port to a suitable module:
+
+```
+# dpdk-devbind -b <module_name> <pci_addr>
+```
+
+### Hugepages Support
+Hugepages support is required for increasing performance. By allocating hugepages, we gain larger pages, using less pages and therefore less TLB misses are involved which speeds up performance. For these reasons, 1GB hugepages will be favored.
+Number of pages can vary according to usage requirements. 
+Hugepages must be marked as shared in order to enable host and guest communication, as packets are transmitted and received on buffers allocated on hugepages memory. 
+ 
+Add engine custom properties for hugepages support (followed by engine restart)
```

Comment:
you should mention `engine-config -g UserDefinedVMProperties` first, to make sure the user does not override something important.